### DISCUSSION TITLE
Why doesn't TextEdit text_changed signal send the new text as a parameter?

### DISCUSSION THREAD

**Discussion by MarcusElg** [godotengine/godot-proposals#4129] [GUI]
LineEdit has the new text as a parameter when sending the text_changed signal which makes it easy to sync it with a variable, this however is not the case with TextEdit. Is there a reason for why? Wouldn't it make more sense for these two nodes to be consistent?

**Comment by Calinou** [comment]
Wouldn't this cause performance issues for very long TextEdits? Therefore, it makes sense for this to be opt-in (i.e. you have to get the text manually).

**Edit:** See also https://github.com/godotengine/godot/issues/59818.

**Reply by MarcusElg** [reply]
That would be the case for very long lineedits as well

**Reply by Calinou** [reply]
It's uncommon for a LineEdit to have very long text, since it's single-line by design. In contrast, long TextEdits are quite common (e.g. code editors).

**Reply by MarcusElg** [reply]
Ok makes sense I guess, would be nice to have the option though.

**Reply by eon-s** [reply]
Maybe it can just send line and column of the changed text? there is another signal that does that.

**Reply by timothyqiu** [reply]
Strings are copy-on-write I believe, so I don't think it's related to text length. Text in LineEdit is a simple String, but text in TextEdit is an array of line info objects that each contains a String of that line. So it'll need to assemble a new String from that array if you want to receive text as parameter.

**Reply by MarcusElg** [reply]
> Strings are copy-on-write I believe, so I don't think it's related to text length. Text in LineEdit is a simple String, but text in TextEdit is an array of line info objects that each contains a String of that line. So it'll need to assemble a new String from that array if you want to receive text as parameter.

Ah ok, that explains it
